### PR TITLE
Add a data map for raid hero gifts

### PR DESCRIPTION
**Pull request by Spinoscythe**
Adds a data map for raid hero gifts

### CODE REVIEW

File: patches/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java.patch
Comment (XFactHD) [inline patches/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java.patch]: You can't assume that a profession that is in the `GIFTS` map also has a data attachment, so you need to split this into multiple if statements and pull the `LootTable` variable up:
```java
LootTable loottable = null;
if (GIFTS.containsKey(villagerprofession)) {
    loottable = p_23010_.level().getServer().getLootData().getLootTable(GIFTS.get(villagerprofession));
} else {
    var gift = net.minecraft.core.registries.BuiltInRegistries.VILLAGER_PROFESSION.wrapAsHolder(villagerprofession).getData(net.neoforged.neoforge.registries.datamaps.builtin.NeoForgeDataMaps.RAID_HERO_GIFTS);
    if (gift != null) {
        loottable = p_23010_.level().getServer().getLootData().getLootTable(gift.lootTable());
    }
}
if (loottable != null) {
    LootParams lootparams = ...;
```

File: src/main/java/net/neoforged/neoforge/registries/datamaps/builtin/NeoForgeDataMaps.java
Comment (XFactHD) [inline src/main/java/net/neoforged/neoforge/registries/datamaps/builtin/NeoForgeDataMaps.java]: The field order in this class and the registration order are intentionally in alphabetical order, please change your additions to match that.

**Review by XFactHD — DISMISSED** [review]

File: patches/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java.patch
Comment (Spinoscythe) [reply patches/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java.patch]: will do, but also is "raid hero gifts" the right terminology?

File: src/main/java/net/neoforged/neoforge/registries/datamaps/builtin/NeoForgeDataMaps.java
Comment (Spinoscythe) [reply src/main/java/net/neoforged/neoforge/registries/datamaps/builtin/NeoForgeDataMaps.java]: fixed!

**Comment by Spinoscythe** [comment]
@XFactHD Added changes

File: patches/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java.patch
Comment (XFactHD) [reply patches/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java.patch]: Sounds fine in my opinion. `Hero of the Village gifts` would technically be the fully correct name since vanilla calls the effect `Hero of the Village`, though that feels too long-winded for my taste.

File: src/main/java/net/neoforged/neoforge/registries/datamaps/builtin/NeoForgeDataMaps.java
Comment (XFactHD) [reply src/main/java/net/neoforged/neoforge/registries/datamaps/builtin/NeoForgeDataMaps.java]: The `RAID_HERO_GIFTS` field needs to be moved above the `VIBRATION_FREQUENCIES` field as well, not just the registration order.

Comment (Spinoscythe) [reply src/main/java/net/neoforged/neoforge/registries/datamaps/builtin/NeoForgeDataMaps.java]: fixed!

File: patches/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java.patch
Comment (XFactHD) [inline patches/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java.patch]: I just noticed that my earlier suggestion is backwards since the datamap should take precedence, that's my mistake 🤦. Please switch around the check to this instead:
```java
LootTable loottable = null;
var gift = net.minecraft.core.registries.BuiltInRegistries.VILLAGER_PROFESSION.wrapAsHolder(villagerprofession).getData(net.neoforged.neoforge.registries.datamaps.builtin.NeoForgeDataMaps.RAID_HERO_GIFTS);
if (gift != null) {
    loottable = p_23010_.level().getServer().getLootData().getLootTable(gift.lootTable());
} else if (GIFTS.containsKey(villagerprofession)) {
    loottable = p_23010_.level().getServer().getLootData().getLootTable(GIFTS.get(villagerprofession));
}
if (loottable != null) {
    ...
```

**Review by XFactHD — COMMENTED** [review]
Looks good to me in terms of the implementation. I'll let someone else take a second look and test it as I'm currently not in a position to test this.

**Comment by Spinoscythe** [comment]
I think you should assign Maty to this

**Comment by TelepathicGrunt** [comment]
There's NeoForge maintainers other than Maty that can review and test this lol. Maty may be too busy as well

**Comment by Spinoscythe** [comment]
sure, I just said Maty because afaik he's the most familiar with data maps

File: src/main/java/net/neoforged/neoforge/registries/datamaps/builtin/NeoForgeDataMaps.java
Comment (Matyrobbrt) [inline src/main/java/net/neoforged/neoforge/registries/datamaps/builtin/NeoForgeDataMaps.java]:
```suggestion
     * <li>{@code loot_table}, the path to the loot table that the villager gives to the player after a raid finishes</li>
```

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]

**Review by Matyrobbrt — APPROVED** [review]

**Comment by Spinoscythe** [comment]
missing a deprecated annotation!
will fix when I come back home

**Pull request merged**
